File: API/Helpers/UserParams.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;

namespace API.Helpers
{
    public class UserParams: PaginationParams
    {
        public string CurrentUsername { get; set; }
        public string Gender { get; set; }
        public int MinAge { get; set; } = 14;
        public int MaxAge { get; set; } = 100;
        public string OrderBy { get; set; } = "lastActive";
    }
}

File: client/src/app/_models/userParams.ts
import { User } from "./user";

export class UserParams {
    gender: string;
    minAge = 14;
    maxAge = 100;
    pageNumber = 1;
    pageSize = 10;
    orderBy = 'lastActive';

    constructor(user: User){
        this.gender = user.gender === 'female' ? 'male' : 'female';
    }
}